runtime/drivers/azure/azure.go

```diff
@@ -0,0 +1,363 @@
+	GlobMaxObjectsMatched int            `mapstructure:"glob.max_objects_matched"`
+	GlobMaxObjectsListed  int64          `mapstructure:"glob.max_objects_listed"`
+	GlobPageSize          int            `mapstructure:"glob.page_size"`
+	url                   *globutil.URL
+	extractPolicy         *rillblob.ExtractPolicy
+}
+
+func parseSourceProperties(props map[string]any) (*sourceProperties, error) {
+	conf := &sourceProperties{}
+	err := mapstructure.Decode(props, conf)
+	if err != nil {
+		return nil, err
+	}
+	if !doublestar.ValidatePattern(conf.Path) {
+		return nil, fmt.Errorf("glob pattern %s is invalid", conf.Path)
+	}
+	bucketURL, err := globutil.ParseBucketURL(conf.Path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse path %q, %w", conf.Path, err)
+	}
+	if bucketURL.Scheme != "azblob" {
+		return nil, fmt.Errorf("invalid scheme %q in path %q", bucketURL.Scheme, conf.Path)
+	}
+
+	conf.url = bucketURL
+	return conf, nil
+}
+
+// getClient returns a new azure blob client.
+func (c *Connection) getClient(ctx context.Context, conf *sourceProperties) (*container.Client, error) {
+	var client *container.Client
+	var accountName, accountKey, sasToken, connectionString string
+	azClientOpts := &container.ClientOptions{}
+
+	if c.config.AllowHostAccess {
+		accountName = os.Getenv("AZURE_STORAGE_ACCOUNT")
+		accountKey = os.Getenv("AZURE_STORAGE_KEY")
+		sasToken = os.Getenv("AZURE_STORAGE_SAS_TOKEN")
+		connectionString = os.Getenv("AZURE_STORAGE_CONNECTION_STRING")
+	} else {
+		accountName = c.config.Account
+		accountKey = c.config.Key
+		sasToken = c.config.SASToken
+		// Added it as per azure blob client implementation, can remove if not required
+		connectionString = c.config.ConnectionString
+	}
+
+	if accountName != "" {
+		svcURL := fmt.Sprintf("https://%s.blob.core.windows.net", accountName)
+		containerURL, err := url.JoinPath(svcURL, conf.url.Host)
+		if err != nil {
+			return nil, err
+		}
+		if accountKey != "" {
+			sharedKeyCred, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+			if err != nil {
+				return nil, fmt.Errorf("failed azblob.NewSharedKeyCredential: %w", err)
+			}
+
+			client, err = container.NewClientWithSharedKeyCredential(containerURL, sharedKeyCred, azClientOpts)
+			if err != nil {
+				return nil, fmt.Errorf("failed container.NewClientWithSharedKeyCredential: %w", err)
+			}
+		} else if sasToken != "" {
+			serviceURL, err := azureblob.NewServiceURL(&azureblob.ServiceURLOptions{
+				AccountName: accountName,
+				SASToken:    sasToken,
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			containerURL, err := url.JoinPath(string(serviceURL), conf.url.Host)
+			if err != nil {
+				return nil, err
+			}
+
+			client, err = container.NewClientWithNoCredential(containerURL, azClientOpts)
+			if err != nil {
+				return nil, fmt.Errorf("failed container.NewClientWithNoCredential: %w", err)
+			}
+		} else {
```

Comment:
1. This case should be outside of the parent if case with a condition that `allow_host_access` is true since if user has done az login they don't need to set `account_name`.
2. As discussed verify if we need to enable `DisableInstanceDiscovery` during `azidentity.NewDefaultAzureCredential`
3. Pass nil instead of `azClientOpts`